schema/workflow.yaml

```diff
@@ -325,357 +394,435 @@ $defs:
                   - title: ScriptExternal
+                    type: object
+                    description: The script's resource.
                     properties:
                       source:
                         $ref: '#/$defs/externalResource'
-                    description: The script's resource.
+                        title: ScriptExternalSource
                     required: [ source ]
                 required: [ language ]
             required: [ script ]
-            description: Enables the execution of custom scripts or code within a workflow, empowering workflows to perform specialized logic, data processing, or integration tasks by executing user-defined scripts written in various programming languages.
           - title: RunShell
+            description: Enables the execution of shell commands within a workflow, enabling workflows to interact with the underlying operating system and perform system-level operations, such as file manipulation, environment configuration, or system administration tasks.
             properties:
               shell:
                 type: object
+                title: Shell
+                description: The configuration of the shell command to run.
                 unevaluatedProperties: false
                 properties:
                   command:
                     type: string
+                    title: ShellCommand
                     description: The shell command to run.
                   arguments:
-                    title: ShellArguments
                     type: object
-                    additionalProperties: true
+                    title: ShellArguments
                     description: A list of the arguments of the shell command to run.
+                    additionalProperties: true
                   environment:
-                    title: ShellEnvironment
                     type: object
-                    additionalProperties: true
+                    title: ShellEnvironment
                     description: A key/value mapping of the environment variables, if any, to use when running the configured process.
+                    additionalProperties: true
                 required: [ command ]
             required: [ shell ]
-            description: Enables the execution of shell commands within a workflow, enabling workflows to interact with the underlying operating system and perform system-level operations, such as file manipulation, environment configuration, or system administration tasks.
           - title: RunWorkflow
+            description: Enables the invocation and execution of nested workflows within a parent workflow, facilitating modularization, reusability, and abstraction of complex logic or business processes by encapsulating them into standalone workflow units.
             properties:
               workflow:
-                title: RunWorkflowDescriptor
                 type: object
+                title: RunWorkflowDescriptor
```

Comment:
If we follow the naming of previous concepts, this should be called `RunWorkflowConfiguration`, `SubflowConfiguration` or something like that. Descriptor is a term we used for runtime expression (WorkflowDescriptor, TaskDescriptor, etc) and might therefore be confusing.

Additionally, I would rename all properties `Subflow...` instead of `RunWorkflow...`.